Python/Product/TestAdapter.Executor/UnitTest/UnitTestTestDiscoverer.cs

```diff
@@ -138,16 +138,22 @@ ITestCaseDiscoverySink discoverySink
 
         public string[] GetArguments(IEnumerable<string> sources, PythonProjectSettings settings, string outputfilename) {
             var arguments = new List<string>();
-            arguments.Add(DiscoveryAdapterPath);
-            arguments.Add("discover");
-            arguments.Add("unittest");
-            arguments.Add("--output-file");
-            arguments.Add(outputfilename);
-            //Note unittest specific options go after this separator
-            arguments.Add("--");
-            arguments.Add(settings.UnitTestRootDir);
-            arguments.Add(settings.UnitTestPattern);
-
+            foreach (var source in sources.ToArray()) {
+                var testRootDir = PathUtils.GetAbsoluteDirectoryPath(settings.ProjectHome, settings.UnitTestRootDir);
+                // only return valid arguments when source is a subpath of test root directory
+                if (PathUtils.IsSamePath(PathUtils.GetParent(source), testRootDir)) {
```

Comment:
This will only work if the source is one directory under the testRootDir. Try putting tests in a deeper folder and I think this will fail.

You might need to use `PathUtils.IsSubpathOf(testRootDir, PathUtils.GetParent(source))` instead

Reply:
I spoke to Stella offline about this and this is an existing bug, even without her changes. She's going to try to fix it as part of this PR because it's so closely related to what this PR is trying to fix. 

But if it becomes too much feature-creep, we'll just create a new issue for it and it can get done later.